outputs.tf

```diff
@@ -24,8 +29,8 @@ output "cluster_version" {
 }
 
 output "cluster_security_group_id" {
-  description = "Security group ID attached to the EKS cluster."
-  value       = local.cluster_security_group_id
+  description = "Cluster security group ID attached to the EKS cluster."
+  value       = var.cluster_version >= 1.14 ? element(concat(aws_eks_cluster.this[*].vpc_config[0].cluster_security_group_id, list("")), 0) : local.additional_security_group_id
```

Comment:
Please don't do this. It will make upgrade paths very scary.

Reply:
this will have no effect on future upgrades.  This problem only exists because node_groups are not backwards compatible.   As soon at 1.16 is released, the possibility to hit this edge case no longer exists and this can be removed.  This only exist because of the issue, which is also broken in the current release.

Reply:
This variable has nothing to do with node groups. You are changing the value of an output based on user supplied input. It is a dangerous pattern.

Nothing is broken in the current release. It just lacks output of the new `vpc_config.cluster_security_group_id` value.